### PR TITLE
added railsy package

### PR DESCRIPTION
**Pull request by mmkaram**
## Description of changes
added railsy package, [homepage](https://mmkaram.github.io/railsy.html)
added mmkaram to maintainer-list
added default.nix to railsy in pkgs/by-name/railsy
added required top level definitions

This is my first attempt at adding a package to nixpkgs, so please let me know if I did anything wrong

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ra/railsy/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "railsy";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "mmkaram";
+    repo = "railsy";
+    rev = "v${version}";
```
Comment (steeleduncan) [inline pkgs/by-name/ra/railsy/default.nix:16]: `refs/tags/v${version}` is safer in the case of conflicting branch name

File: pkgs/by-name/ra/railsy/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    openssl
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  PKG_CONFIG_PATH = "${openssl.dev}/lib/pkgconfig";
+
+  meta = with lib; {
```
Comment (steeleduncan) [inline pkgs/by-name/ra/railsy/default.nix:34]: `with lib` is deprecated #208242

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Comment by steeleduncan** [comment]
Please squash the commits. There should be one commit adding yourself to the maintainers, and one adding the package (with commit message `railsy: add at 0.1.1`)

**Comment by mmkaram** [comment]
> Please squash the commits. There should be one commit adding yourself to the maintainers, and one adding the package (with commit message `railsy: add at 0.1.1`)

I think I added one extra commit in the squash, so I'll try my hand at all this a second time

**Pull request closed without merging**